packages/core/src/runtime/compileCSS.ts

```diff
@@ -60,61 +60,69 @@ export function compileCSSRules(cssRules: string): string[] {
   return rules;
 }
 
-export function compileCSS(options: CompileCSSOptions): [string /* ltr definition */, string? /* rtl definition */] {
-  const { className, media, layer, pseudo, support, property, rtlClassName, rtlProperty, rtlValue, value } = options;
+function createCSSRule(classNameSelector: string, cssDeclaration: string, pseudos: string[]): string {
+  let globalSelector = '';
+  let cssRule = cssDeclaration;
 
-  const classNameSelector = `.${className}`;
-  const cssDeclaration = Array.isArray(value)
-    ? `{ ${value.map(v => `${hyphenateProperty(property)}: ${v}`).join(';')}; }`
-    : `{ ${hyphenateProperty(property)}: ${value}; }`;
+  if (pseudos.length > 0) {
+    cssRule = pseudos.reduceRight((acc, selector, index) => {
+      // Should be handled by namespace plugin of Stylis, is buggy now
+      // Issues are reported:
+      // https://github.com/thysultan/stylis.js/issues/253
+      // https://github.com/thysultan/stylis.js/issues/252
+      if (selector.indexOf(':global(') === 0) {
+        // 👇 :global(GROUP_1)GROUP_2
+        const GLOBAL_PSEUDO_REGEX = /global\((.+)\)(.+)?/;
+        const result = GLOBAL_PSEUDO_REGEX.exec(selector)!;
 
-  let rtlClassNameSelector: string | null = null;
-  let rtlCSSDeclaration: string | null = null;
+        globalSelector = result[1] + ' ';
+        const restPseudo = result[2] || '';
 
-  if (rtlProperty && rtlClassName) {
-    rtlClassNameSelector = `.${rtlClassName}`;
-    rtlCSSDeclaration = Array.isArray(rtlValue)
-      ? `{ ${rtlValue.map(v => `${hyphenateProperty(rtlProperty)}: ${v}`).join(';')}; }`
-      : `{ ${hyphenateProperty(rtlProperty)}: ${rtlValue}; }`;
-  }
+        // should be normalized to handle ":global(SELECTOR) &"
+        const normalizedPseudoSelector = normalizePseudoSelector(restPseudo);
 
-  let cssRule = '';
+        if (normalizedPseudoSelector === '') {
+          return acc;
+        }
 
-  // Should be handled by namespace plugin of Stylis, is buggy now
-  // Issues are reported:
-  // https://github.com/thysultan/stylis.js/issues/253
-  // https://github.com/thysultan/stylis.js/issues/252
-  if (pseudo.indexOf(':global(') === 0) {
-    // 👇 :global(GROUP_1)GROUP_2
-    const GLOBAL_PSEUDO_REGEX = /global\((.+)\)(.+)?/;
-    const [, globalSelector, restPseudo = ''] = GLOBAL_PSEUDO_REGEX.exec(pseudo)!;
+        return `${normalizedPseudoSelector} { ${acc} }`;
+      }
```

Comment:
I moved the hack with small changes, will replace it with a Stylis plugin in a follow up PR.